src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
+      else
+        # no metadata.json file, so all we have is `name`
+        module_data[:name] = mod.name
+        module_data[:version] = nil
```

Comment:
Not a big deal but could do something shorter like the following to avoid having any conditional logic for the `:version` field and still get the behavior of a `nil` being returned when no version can be derived from a metadata.json file for the module:

```ruby
  def self.getModules(env)
    env.modules.collect do |mod|
      {:name => mod.forge_name ||= mod.name,
       :version => mod.version}
    end
  end
```

That said, though, I'm good with what you have already if you want to go with that instead.